test/workers/scanner_test.go

```diff
@@ -48,8 +56,9 @@ func TestScanner(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	// Sleep for two cycles.
-	time.Sleep(2 * workers.SleepBetweenScans())
+
+	// Sleep for a while, giving a chance to the scanner to pick the skylink up.
+	time.Sleep(cyclesToWait * workers.SleepBetweenScans())
```

Comment:
I would not use `workers.SleepBetweenScans` considering that it uses some randomness itself.
Instead I'd just define a `maxSleepBetweenScans` which is the largest possible outcome of `SleepBetweenScans` and sleep 2 times that. That should be plenty of time.

Reply:
Doing this in a retry would be misleading. When using retry we expect the state to change to the expected value at some point. In this case we want to ensure that no matter how long we wait the state will not change. That's why I'm waiting for the maximum allowed time for which the retry calls are running. This is similar to Chris's idea of `maxSleepBetweenScans` but it's also different because it's a lot longer. The shorter approach that he suggested just didn't cut it.